src/Security/JsonWebTokenAuthenticator.php

```diff
@@ -38,29 +33,38 @@ public function __construct(JsonWebTokenManager $jwtManager)
     /**
      * @inheritdoc
      */
-    public function createToken(Request $request, $providerKey)
+    public function start(Request $request, AuthenticationException $authException = null)
+    {
+        return new Response('Auth header required', 401);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function supports(Request $request)
     {
-        $jwt = false;
+        return $request->headers->has('X-JWT-Authorization');
+    }
 
+    /**
+     * @inheritdoc
+     */
+    public function getCredentials(Request $request)
+    {
         $authorizationHeader = $request->headers->get('X-JWT-Authorization');
         if (preg_match('/^Token (.*)$/', $authorizationHeader, $matches)) {
-            $jwt = $matches[1];
-        }
-        if (!$jwt) {
-            throw new AuthenticationCredentialsNotFoundException('No JSON Web Token was found in the request');
+            return $matches[1];
         }
 
-        return new PreAuthenticatedToken(
-            'anon.',
-            $jwt,
-            $providerKey
-        );
+        return null;
```

Comment:
according to the should [docs](https://api.symfony.com/3.4/Symfony/Component/Security/Guard/AuthenticatorInterface.html#method_getCredentials), this should throw `UnexpectedValueException` instead of returning `null` if no credentials could be found.